cmake/SetupCompilerFlags.cmake

```diff
@@ -21,3 +21,10 @@ endif ()
 if (ENABLE_HIP)
 	set(HIP_HIPCC_FLAGS "${HIP_HIPCC_FLAGS} -Wno-inconsistent-missing-override")
 endif()
+
+blt_append_custom_compiler_flag(
+  FLAGS_VAR UMPIRE_PEDANTIC_FLAG
+  DEFAULT  "-Wpedantic"
```

Comment:
Do all compilers use this flag? Should an option for pedantic warnings be added to BLT directly?

Reply:
Possibly not, that is why I started this "draft" PR.  I wanted to run it through our CI.  In parallel I am also setting up to run with our other compilers on our HPC systems.

Reply:
I cannot answer the BLT question.  Perhaps @davidbeckingsale has an opinion on that?

Reply:
I think we should guard this with a CMake option for now. E.g. `ENABLE_PEDANTIC_WARNINGS`

Reply:
Done.

Reply:
Ace, thanks!